R/pkg/R/sparkR.R

```diff
@@ -334,17 +334,6 @@ sparkR.session <- function(
     assign(".sparkRsession", sparkSession, envir = .sparkREnv)
   }
 
-  # Check if version number of SparkSession matches version number of SparkR package
-  jvmVersion <- callJMethod(sparkSession, "version")
-  # Remove -SNAPSHOT from jvm versions
-  jvmVersionStrip <- gsub("-SNAPSHOT", "", jvmVersion)
-  rPackageVersion <- paste0(packageVersion("SparkR"))
-
-  if (jvmVersionStrip != rPackageVersion) {
-    warning(paste("Version mismatch between Spark JVM and SparkR package. JVM version was",
-                  jvmVersion, ", while R package version was", rPackageVersion))
-  }
-
   sparkSession
```

Comment:
Alright, I checked previous Spark preview (https://archive.apache.org/dist/spark/spark-2.0.0-preview/):

```
Type: Package
Title: R frontend for Spark
Version: 2.0.0
Date: 2013-09-09
```

The version was 2.0.0.

PySpark version was added as of Spark 2.1 (https://issues.apache.org/jira/browse/SPARK-18129). I think 3.0.0.dev0 makes sense for now. Seems like we can increase it to dev1 later per PEP 440 (https://www.python.org/dev/peps/pep-0440/).